pkg/apis/application/v1alpha1/types.go

```diff
@@ -432,6 +432,33 @@ type ApplicationSourceKustomize struct {
 	ForceCommonLabels bool `json:"forceCommonLabels,omitempty" protobuf:"bytes,7,opt,name=forceCommonLabels"`
 	// ForceCommonAnnotations specifies whether to force applying common annotations to resources for Kustomize apps
 	ForceCommonAnnotations bool `json:"forceCommonAnnotations,omitempty" protobuf:"bytes,8,opt,name=forceCommonAnnotations"`
+	// Replicas is a list of Kustomize Replicas override specifications
+	Replicas KustomizeReplicas `json:"replicas,omitempty" protobuf:"bytes,9,opt,name=replicas"`
+}
+
+type KustomizeReplica struct {
+	// Name of Deployment or StatefulSet
+	Name string `json:"name" protobuf:"bytes,1,name=name"`
+	// Number of replicas
+	Count int64 `json:"count" protobuf:"bytes,2,name=count"`
```

Comment:
I wonder if we could make this an `IntOrString`. That would make it templateable in ApplicationSets.

Reply:
Thank you for your review!
* I changed to IntOrString and added some tests.
* Updated PR description to include a manual test for ApplicationSets.